File: mycuration/src/main/java/com/phicdy/mycuration/domain/filter/FilterTask.java
package com.phicdy.mycuration.domain.filter;

import com.phicdy.mycuration.data.db.DatabaseAdapter;
import com.phicdy.mycuration.data.filter.Filter;

import java.util.ArrayList;
  
public class FilterTask {

    private final DatabaseAdapter dbAdapter;
      
    public FilterTask() {
        dbAdapter = DatabaseAdapter.getInstance();
    }
  
      
    public void applyFiltering(int feedId) {
        //Get Filters of Feed ID
        ArrayList<Filter> filterList = dbAdapter.getEnabledFiltersOfFeed(feedId);
        if (filterList == null || filterList.size() == 0) return;
        dbAdapter.applyFiltersOfFeed(filterList, feedId);
    }
      
 
      
      
}

File: mycuration/src/main/java/com/phicdy/mycuration/domain/filter/FilterTask.kt
package com.phicdy.mycuration.domain.filter

import com.phicdy.mycuration.data.repository.ArticleRepository
import com.phicdy.mycuration.data.repository.FilterRepository
import kotlinx.coroutines.coroutineScope

class FilterTask(private val articleRepository: ArticleRepository,
                 private val filterRepository: FilterRepository) {

    suspend fun applyFiltering(feedId: Int): Int = coroutineScope {
        val filters = filterRepository.getEnabledFiltersOfFeed(feedId)
        if (filters.size == 0) return@coroutineScope 0
        return@coroutineScope articleRepository.applyFiltersOfRss(filters, feedId)
    }
}
